protocol/src/main/java/org/fido/iot/protocol/epid/EpidSignatureVerifier.java

```diff
@@ -0,0 +1,105 @@
+        throw new RuntimeException();
+      }
+      int sgType = sigInfoAComposite.getAsNumber(Const.FIRST_KEY).intValue();
+      byte[] groupId = sigInfoAComposite.getAsBytes(Const.SECOND_KEY);
+      byte[] msg = createEpidSignatureBodyMessage(signatureComposite, groupId, sgType);
+      String url = null;
+      String path;
+      switch (sgType) {
+        case Const.SG_EPIDv10:
+        case Const.SG_EPIDv11:
+          path = Joiner.on(Const.URL_PATH_SEPARATOR).join(Const.EPID_PROTOCOL_VERSION_V1,
+              Const.EPID_11, Const.EPID_PROOF_URI_PATH);
+          url = new URL(EpidUtils.getEpidOnlineUrl().toString()).toURI().resolve(path).toString();
+          break;
+        case Const.SG_EPIDv20:
+          path = Joiner.on(Const.URL_PATH_SEPARATOR).join(Const.EPID_PROTOCOL_VERSION_V1,
+              Const.EPID_11, Const.EPID_PROOF_URI_PATH);
+          url = new URL(EpidUtils.getEpidOnlineUrl().toString()).toURI().resolve(path).toString();
+          break;
+        default:
+          throw new IOException("Invalid sgType");
+      }
+      byte[] response = EpidHttpClient.doPost(url, msg);
+      if (null != response && response.length == 0) {
+        return true;
+      } else {
+        return false;
+      }
+    } catch (IOException | URISyntaxException ex) {
+      return false;
+    }
+  }
+
+  private static byte[] createEpidSignatureBodyMessage(Composite signatureComposite, byte[] groupId,
+      int sgType) throws IOException {
+    byte[] signature = signatureComposite.getAsBytes(Const.COSE_SIGN1_SIGNATURE);
+    byte[] signedPayload = createEpidPayload(signatureComposite, sgType);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    baos.write(groupId);
+    baos.write(ByteBuffer.allocate(2).order(ByteOrder.BIG_ENDIAN)
+        .putShort((short) signedPayload.length).array());
+    baos.write(signedPayload);
+    baos.write(ByteBuffer.allocate(2).order(ByteOrder.BIG_ENDIAN).putShort((short) 0).array());
+    baos.write(signature);
+    return baos.toByteArray();
+  }
+
+  // Generate the signed Payload depnding on the sgType.
+  private static byte[] createEpidPayload(Composite signatureBody, int sgType) throws IOException {
+    Composite uph = signatureBody.getAsComposite(Const.COSE_SIGN1_UNPROTECTED);
+    Composite payload = signatureBody.getAsComposite(Const.COSE_SIGN1_PAYLOAD);
+    byte[] payloadAsBytes = signatureBody.getAsBytes(Const.COSE_SIGN1_PAYLOAD);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    switch (sgType) {
+      case Const.SG_EPIDv10:
+        baos.write((byte) uph.getAsBytes(Const.EAT_MAROE_PREFIX).length);
+        baos.write(uph.getAsBytes(Const.EAT_MAROE_PREFIX));
+        baos.write(payload.getAsBytes(Const.EAT_NONCE));
+        baos.write(payloadAsBytes);
+        break;
+      case Const.SG_EPIDv11:
+        baos.write(ByteBuffer.allocate(48).put(4, (byte) 0x48).put(8, (byte) 0x08).array());
```

Comment:
Is the 48, 0x48 thing correct?  It looks suspect, like it might be a base error.

Reply:
As per the protocol spec, we put values 0x48 and 0x08 as a single byte. And we've bee doing this is previous iterations.